File: archive/k/kotlin/PrimeNumber.kt
fun main(args: Array<String>) 
{
    var num = readLine()!!.toIntOrNull()
    if(num == null){
        print("Usage: please input a non-negative integer")
        return
    }
    if(num>1)
    {
        for(i in 2 until num)
        {
            if(num%i == 0)
            {
                println("Composite")
                return
            }
        }
        println("Prime")
    }
    else
    {
        println("Composite")
    }
}

File: archive/j/julia/prime-number.jl
function err() 
    return "Usage: please input a non-negative integer"
 end
 

function isPrime(num)
# handle 0 and 1
    if num <= 1 
        return "composite"
    end 

# default handler
    for i in 2:sqrt(num) #loop starting at 2, ending at the squareroot of the input
        if (num % i == 0)
            return "composite"
        end
    end
    return "prime"             
end

#check for a valid input
try
    n = parse(Int, ARGS[1])
    if n < 0
        println(err())
    else
        println(isPrime(n))
    end
catch e
    println(err())
end
